### ISSUE TITLE
[consistent-type-assertions] assertionStyle doesn't enforces <const> case

### ISSUE THREAD

**Issue by gtkatakura** [typescript-eslint/typescript-eslint#4684]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```tsx
const example = <const>{ key: 'value' }
```

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-assertions": ["error", {"assertionStyle":"as"}]
  }
}
```

**Expected Result**

Should complains about the const forcing to use `as const`.

**Actual Result**

`assertionStyle` doesn't has impact over `<const>` or `as const` today

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.15.0` |
| `@typescript-eslint/parser`        | `5.15.0` |
| `TypeScript`                       | `3.9.10` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.6` |

**Comment by bradzacher** [comment]
I'm not sure I understand - why do you think that `as const` shouldn't be allowed?
`as const` is always safe to do because it only ever narrows the type.

**Comment by gtkatakura** [comment]
I am talking about the styling. It should enforces me to use `as const` instead `<const>` because I defined the `assertionStyle` as `as`.
